### PR TITLE
Support WTForms 3

### PR DESCRIPTION
**Pull request by Glandos**
This bring the latest version, **but** some breaking changes are introduced.
Especially, StringField now default to `None` instead of `''`, so I changed the tests, without knowing the real impact.

@zorun do you have an idea of a real use case where it can break? All columns are nullable, as far as I know, but I want to be extra sure.

### CODE REVIEW

**Comment by Glandos** [comment]
Ah, of course, this breaking change cannot be backported, at least in tests… I'll see it later, then.

File: setup.cfg [hunk 1]
```diff
@@ -35,7 +35,7 @@ install_requires =
     Flask-SQLAlchemy>=2.4,<3
     Flask-Talisman>=0.8,<1
     Flask-WTF>=0.14.3,<2
-    WTForms>=2.3.1,<2.4
+    WTForms>=2.3.1,<3.1
```
Comment (almet) [inline setup.cfg:38]: Shouldn't this be `>=3.0` ?

File: ihatemoney/forms.py [hunk 2]
```diff
@@ -312,6 +326,7 @@ class BillForm(FlaskForm):
     original_currency = SelectField(_("Currency"), validators=[DataRequired()])
     external_link = URLField(
         _("External link"),
+        default="",
```
Comment (almet) [inline ihatemoney/forms.py:329]: So it means that by default it stores an empty string?

File: ihatemoney/forms.py [hunk 3]
```diff
@@ -9,9 +9,23 @@
...
+    Label,
+    PasswordField,
+    SelectField,
+    SelectMultipleField,
+    StringField,
+    SubmitField,
+)
+
+try:
+    # Compat for WTForms <= 2.3.3
```
Comment (almet) [inline ihatemoney/forms.py:25]: Why should we keep backwards compatibility?

File: ihatemoney/forms.py (hunk 2 above)
Comment (almet) [reply ihatemoney/forms.py:329]: Disregard, I understand now :-)

**Review by almet — APPROVED** [review]

File: ihatemoney/forms.py (hunk 3 above)
Comment (Glandos) [reply ihatemoney/forms.py:25]: That's exactly my question. My experience showed me that jumping early on the "new backward incompatible release" train can be really annoying. So I tend to prefer solutions that introduce a backward compatibility layer, but I think this is not possible here, due to the default change for string values.

Comment (almet) [reply ihatemoney/forms.py:25]: > My experience showed me that jumping early on the "new backward incompatible release" train can be really annoying.

Ah, that seems bad indeed. Could you explain why?

**Comment by symphorien** [comment]
can this be merged ? I kind of need that for the downstream nixos package.

**Comment by almet** [comment]
Okay, let's do that for now, and drop it later on :-)

**Comment by symphorien** [comment]
thanks !

**Pull request merged**